### ISSUE TITLE
fails to start on file with illegal name in playlist (saved tab): UnicodeDecodeError

### ISSUE THREAD

**Issue by genodeftest** [exaile/exaile#38]
Steps to reproduce:
1. start exaile
2. add a huge collection of files to a tab
3. close exaile. This playlist wil be saved in ~/.local/exaile/saved_tabs/
4. start exaile

Exaile fails to start. This is the backtrace:
~~~
$ exaile
INFO    : Loading Exaile 3.4.3+1c3041f on Python 2.7.8...
INFO    : Loading settings...
INFO    : Using de_DE UTF-8 locale
** Message: pygobject_register_sinkfunc is deprecated (GstObject)
INFO    : Loading plugins...
INFO    : Loading collection...
INFO    : Loading devices...
INFO    : Connected to udisks2
INFO    : Loading interface...
INFO    : Using GTK+ 2.24.25
INFO    : Loading main window...
Traceback (most recent call last):
  File "/path/to/exaile/exaile.py", line 85, in <module>
    main()
  File "/path/to/exaile/exaile.py", line 82, in main
    exaile = main.Exaile()
  File "/path/to/exaile/xl/main.py", line 167, in __init__
    self.__init()
  File "/path/to/exaile/xl/main.py", line 325, in __init
    self.gui = xlgui.Main(self)
  File "/path/to/exaile/xlgui/__init__.py", line 122, in __init__
    self.main = main.MainWindow(self, self.builder, exaile.collection)
  File "/path/to/exaile/xlgui/main.py", line 141, in __init__
    self._setup_widgets()
  File "/path/to/exaile/xlgui/main.py", line 232, in _setup_widgets
    self.playlist_container = PlaylistContainer('saved_tabs', player.PLAYER)
  File "/path/to/exaile/xlgui/playlist_container.py", line 495, in __init__
    self.notebooks.append(PlaylistNotebook(manager_name, player, '<Control><Shift>t'))
  File "/path/to/exaile/xlgui/playlist_container.py", line 161, in __init__
    self.load_saved_tabs()
  File "/path/to/exaile/xlgui/playlist_container.py", line 264, in load_saved_tabs
    self.create_tab_from_playlist(pl)
  File "/path/to/exaile/xlgui/playlist_container.py", line 184, in create_tab_from_playlist
    page = PlaylistPage(playlist, self.player)
  File "/path/to/exaile/xlgui/widgets/playlist.py", line 390, in __init__
    self.view = PlaylistView(playlist, player)
  File "/path/to/exaile/xlgui/widgets/playlist.py", line 615, in __init__
    self._setup_columns()
  File "/path/to/exaile/xlgui/widgets/playlist.py", line 759, in _setup_columns
    self.model = PlaylistModel(self.playlist, columns, self.player)
  File "/path/to/exaile/xlgui/widgets/playlist.py", line 1225, in __init__
    self.on_tracks_added(None, self.playlist, enumerate(self.playlist)) # populate the list
  File "/path/to/exaile/xlgui/widgets/playlist.py", line 1306, in on_tracks_added
    self.insert(position, self.track_to_row_data(track, position))
  File "/path/to/exaile/xlgui/widgets/playlist.py", line 1274, in track_to_row_data
    return [track, self.icon_for_row(position)] + [providers.get_provider('playlist-columns', name).formatter.format(track) for name in self.columns]
  File "/path/to/exaile/xl/formatter.py", line 402, in format
    substitute = track.get_tag_display(tag)
  File "/path/to/exaile/xl/trax/track.py", line 631, in get_tag_display
    value = unicode(value)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x91 in position 9: invalid start byte
~~~

I don't know for sure but I think this is related to some file in my music library.
As a workaround deleting the ~/.local/share/exaile/saved_tabs/ folder works fine and stops the bug from recurring until adding the same file(s) again.

**Comment by genodeftest** [comment]
This bug happens when step 2 contains a file with incorrect encoded name to the playlist. In my case the file:/// URL in saved_tabs playlist is containing a `%91` escape sequence. The file browser tells me that this file has an incorrect encoded name (so does nautilus). gnome-terminal and several other applications show this too.

When trying to add this file to exaile's playlist (tab) on the GUI nothing happens. In background this file is added to the playlist file.

When running `exaile --debug` and repeating step 2 for a file with incorrect encoded name I get this traceback:
~~~
Traceback (most recent call last):
  File "/path/to/exaile/xl/event.py", line 264, in emit
    event.data, *cb.args, **cb.kwargs)
  File "/path/to/exaile/xlgui/widgets/playlist.py", line 1306, in on_tracks_added
    self.insert(position, self.track_to_row_data(track, position))
  File "/path/to/exaile/xlgui/widgets/playlist.py", line 1274, in track_to_row_data
    return [track, self.icon_for_row(position)] + [providers.get_provider('playlist-columns', name).formatter.format(track) for name in self.columns]
  File "/path/to/exaile/xl/formatter.py", line 402, in format
    substitute = track.get_tag_display(tag)
  File "/path/to/exaile/xl/trax/track.py", line 631, in get_tag_display
    value = unicode(value)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x91 in position 9: invalid start byte
 (xl.event)
~~~
which is the same position as in the above backtrace.

I think there are two options:
1) exaile should handle this file like any ordinary file. I don't know how hard that is.
2) exaile should warn the user when he tries to add this file to his library and/or playlist.

**Comment by Zarokka** [comment]
I had a similar issue.
For me it was an unicode character in the track name "No One´s Watching" where the apostrophe is u'\xb4'.
This could be fixed by changing "str" to "unicode in line (branch: gi file: playlist.py:527) to:
ar = [self.track_image, unicode(wrapper), wrapper]

But i don't know python very well and i don't know if it is a proper fix, and there may be other places where the same problem exists.

**Comment by sjohannes** [comment]
@Zarokka Your error seems to be different to the original bug (which *I think* is fixed in `gi`). Could you post the traceback + error message that Exaile displays in the console?

**Comment by Zarokka** [comment]
Here is the stacktrace:

INFO    : Loading panels...
Traceback (most recent call last):
  File "/home/martin/develop/git/exaile/exaile.py", line 85, in <module>
    main()
  File "/home/martin/develop/git/exaile/exaile.py", line 82, in main
    exaile = main.Exaile()
  File "/home/martin/develop/git/exaile/xl/main.py", line 168, in __init__
    self.__init()
  File "/home/martin/develop/git/exaile/xl/main.py", line 338, in __init
    self.gui = xlgui.Main(self)
  File "/home/martin/develop/git/exaile/xlgui/__init__.py", line 130, in __init__
    self.panel_notebook = panels.PanelNotebook(exaile, self)
  File "/home/martin/develop/git/exaile/xlgui/panels.py", line 97, in __init__
    _register_builtin_panels(exaile, gui.main.window)
  File "/home/martin/develop/git/exaile/xlgui/panels.py", line 284, in _register_builtin_panels
    'playlists')
  File "/home/martin/develop/git/exaile/xlgui/panel/playlists.py", line 622, in __init__
    self._load_playlists()
  File "/home/martin/develop/git/exaile/xlgui/panel/playlists.py", line 695, in _load_playlists
    self._load_playlist_nodes(playlist)
  File "/home/martin/develop/git/exaile/xlgui/panel/playlists.py", line 527, in _load_playlist_nodes
    ar = [self.track_image, str(wrapper), wrapper]
UnicodeEncodeError: 'ascii' codec can't encode character u'\xb4' in position 7: ordinal not in range(128)
Exception TypeError: "'NoneType' object is not callable" in  ignored
Exception TypeError: "'NoneType' object is not callable" in  ignored

**Comment by sjohannes** [comment]
Thanks; I've committed the change, along with some changes to the surrounding code to make the intention clearer. You may get a conflict due to a variable name change.

(Regarding the original bug, I have to test whether it's still present in `gi`.)

**Comment by Zarokka** [comment]
I tested with your changes and it solved my issue.
I also tested with unicode in filenames (what the original bug seemed to be about) and that works fine too.
Thanks.

**Comment by genodeftest** [comment]
I can confirm this issue is fixed with latest version 4.0.0dev+d6ead2c from gi branch.
